internal/nodes/ingress_model_rel.go

```diff
@@ -307,10 +313,28 @@ func IngressChanges(ingName string, namespace string, key string) ([]string, boo
 			objects.SharedSvcLister().IngressMappings(metav1.NamespaceAll).RemoveIngressClassMappings(namespace + "/" + ingName)
 		}
 
-		services := parseServicesForIngress(ingObj.Spec, key)
-		for _, svc := range services {
-			utils.AviLog.Debugf("key: %s, msg: updating ingress relationship for service: %s", key, svc)
+		_, oldSvcs := objects.SharedSvcLister().IngressMappings(namespace).GetIngToSvc(ingName)
+		currSvcs := parseServicesForIngress(ingObj.Spec, key)
+
+		svcToDel := lib.Difference(oldSvcs, currSvcs)
+		for _, svc := range svcToDel {
+			_, ingrforSvc := objects.SharedSvcLister().IngressMappings(namespace).GetSvcToIng(svc)
+			ingrforSvc = utils.Remove(ingrforSvc, ingName)
+			//objects.SharedSvcLister().IngressMappings(namespace).UpdateSvcToIngMapping(svc, ingrforSvc)
+			if len(ingrforSvc) == 0 {
```

Comment:
should we be checking for AutoAnnotateNPLSvc() here as well? also let's remove the commented out code above.

Reply:
Good catch ! Added the condition.